### DISCUSSION TITLE
Add a mechanism to avoid having to hardcode DataScheme on IntentFilter

### DISCUSSION THREAD

**Discussion by efonsecab** [dotnet/maui#12590] [Ideas]
When using MSAL Authentication one of the recommended approaches for the Android app is to create a custom activity class and mark it with an IntentFilter attribute, where the DataScheme would have the MSAL Id of the Azure App. This however force you to hardcode the id.

This is a sample of the code I would like to keep clean of Ids
https://github.com/pticostaricags/FairPlaySocial/blob/main/src/FairPlaySocialSln/FairPlaySocial.MAUIBlazor/Platforms/Android/MsalActivity.cs

I'm not sure what would be the best way to solve this of if even possible. Exposing an interface with the methods exposing the respective context, as a parameter, and then assign the DataScheme on that context in the method's implementation?

**Comment by maschroeder-z** [comment]
Is there any other option to use the MSAL with the MsalActivity approach? How it is possible to distribute a MAUI based APP for different customers when the Cllient-ID (Application-ID) must be fix coded? I want to access Office 365 data and each customer has it's own App-Register in Entra ...
